packages/contracts-core/contracts/test/utils/NomadCore.sol

```diff
@@ -0,0 +1,340 @@
+        bytes memory data;
+        updaterManager = new UpdaterManager(updater);
+        updaterManagers.push(updaterManager);
+        assertEq(updaterManager.updater(), updater);
+
+        homeImplementation = new Home(localDomain);
+        homeController = new UpgradeBeaconController();
+        homeBeacon = new UpgradeBeacon(
+            address(homeImplementation),
+            address(homeController)
+        );
+        data = abi.encodeWithSignature(
+            "initialize(address)",
+            address(updaterManager)
+        );
+        homeProxy = new UpgradeBeaconProxy(address(homeBeacon), data);
+        home = Home(address(homeProxy));
+
+        updaterManager.setHome(address(home));
+
+        homes.push(home);
+        homeProxies.push(homeProxy);
+        homeImplementations.push(homeImplementation);
+        homeBeacons.push(homeBeacon);
+        homeControllers.push(homeController);
+
+        xAppConnectionManager = new XAppConnectionManager();
+        xAppConnectionManager.setHome(address(home));
+        xAppConnectionManagers.push(xAppConnectionManager);
+
+        governanceRouterImplementation = new GovernanceRouter(
+            localDomain,
+            recoveryTimelock
+        );
+        governanceRouterController = new UpgradeBeaconController();
+        governanceRouterBeacon = new UpgradeBeacon(
+            address(governanceRouterImplementation),
+            address(governanceRouterController)
+        );
+        data = abi.encodeWithSignature(
+            "initialize(address,address)",
+            address(xAppConnectionManager),
+            recoveryManager
+        );
+        governanceRouterProxy = new UpgradeBeaconProxy(
+            address(governanceRouterBeacon),
+            data
+        );
+        // Create a GovernaceRouter object so that we can call the Proxie's functions
+        // as if we would call the contract itself.
+        governanceRouter = GovernanceRouter(address(governanceRouterProxy));
+
+        governanceRouters.push(governanceRouter);
+        governanceRouterProxies.push(governanceRouterProxy);
+        governanceRouterImplementations.push(governanceRouterImplementation);
+        governanceRouterBeacons.push(governanceRouterBeacon);
+        governanceRouterControllers.push(governanceRouterController);
+
+        domainToIndex[localDomain] = homes.length - 1;
+    }
+
+    function createLocalReplicaForRemoteDomain(
+        uint32 localDomain,
+        uint32 remoteDomain
+    ) public {
+        bytes memory data;
+        replicaImplementation = new Replica(localDomain);
+        replicaController = new UpgradeBeaconController();
+        replicaBeacon = new UpgradeBeacon(
+            address(replicaImplementation),
+            address(replicaController)
+        );
+        data = abi.encodeWithSignature(
+            "initialize(uint32,address,bytes32,uint256)",
+            remoteDomain,
+            updater,
+            committedRoot,
```

Comment:
hmm okay so every Replica will have the same `committedRoot` and the same `updater`

Reply:
this seems vaguely problematic to me

Reply:
Why? This is a new nomad protocol deployment from the start, aiming to make it as simple as possible to test on the protocol. 

Having different updaters doesn't offer anything to the user?